data/transactions/verify/txn.go

```diff
@@ -235,12 +244,12 @@ func stxnVerifyCore(s *transactions.SignedTxn, txnIdx int, groupCtx *GroupContex
 			batchVerifier); ok {
 			return nil
 		}
-		return errors.New("multisig validation failed")
+		return signatureErrorf("multisig validation failed")
 	}
 	if hasLogicSig {
 		return logicSigBatchVerify(s, txnIdx, groupCtx, batchVerifier)
```

Comment:
@jdtzmn and reviewers - Posting a question having done effectively no research to see if others know off-hand:  When invoking `logicSigBatchVerify`, the argument `batchVerifier` is _not_ used.  Is it intentional?

Having not yet reviewed git history, it's unclear to me if (a) parameter ought to be removed, (b) there's a regression and parameter _should_ be used, or (c) something else.

Reply:
I suspect the intent was to use it in the case that logisig is a delegated logisig (those have crypto signatures on them. a normal logicsig does not). It looks to my untrained eye that we don't have a bug - the delegated logicsig is handled with it's own batchverifier. But it is maybe a lost optimization - we don't batch those signatures with the normal top-level signatures.  @algoidan could comment more.

Reply:
I think you are right. The idea was to batch as many signatures as we can.  In this specific case, I think that this parameter can be removed.

Reply:
@algoidan Thanks for the clarification.

Resolving the thread - I opened https://github.com/algorand/go-algorand/pull/4360 to continue discussion.  I'll request review once CI passes.